src/App.tsx

```diff
@@ -47,13 +47,50 @@ const useAppStyles = makeStyles(theme => {
 const App = (): JSX.Element => {
   const classes = useAppStyles(theme);
 
-  // This Hook is used to keep track of whether to render the table or CommitmentDetails
-  // const [displayDetails, setDisplayDetails] = React.useState(false);
   // This Hook is used to keep track of which CommitmentDetails should be displayed
   const [displayId, setDisplayId] = React.useState(
     window.location.hash.slice(1) || ''
   );
 
+  // This is intended to be used for more generalized purposes
+  const [history, setHistory] = React.useState([] as string[]); // May not need to be a hook
+
+  const isInDoc = useRef(true);
```

Comment:
This isn't really what a ref is used for.  The purpose of a ref is to assign it to a react component so that you can reference the DOM element it generates later.  A common use case is the `<canvas />` html element.  You would say something like

````
  // this code draws a red rectangle atop a blue rectangle on a 100px by 100px canvas
  const canvasRef = useRef(null);
  useEffect(() => {
    const canvas = canvasRef.current;
    const ctx = canvas.getContext('2d');
    ctx.fillStyle = 'blue';
    ctx.fillRect(10, 10, 50, 50);

    ctx.fillStyle = 'red';
    ctx.fillRect(20, 20, 25, 25);
  });

  return <canvas ref={canvasRef} width="100" height="100" />;
````

Here's a jsfiddle for it https://jsfiddle.net/7orfw9a2/